x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorConfigurationAction.java

```diff
@@ -73,8 +83,15 @@ public ActionRequestValidationException validate() {
                 validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
             }
 
-            if (Objects.isNull(configuration)) {
-                validationException = addValidationError("[configuration] cannot be [null].", validationException);
+            if (configuration == null && configurationValues == null) {
+                validationException = addValidationError("[configuration] and [values] cannot both be null.", validationException);
+            }
+
+            if (configuration != null && configurationValues != null) {
+                validationException = addValidationError(
+                    "[configuration] and [values] cannot both be provided in the same request.",
+                    validationException
+                );
```

Comment:
Nice 👏  was tempted to ask the question in the index service above, but wanted to check first, whether you're checking it in the request layer!